packages/app/src/components/theme/SwatchesGrid.tsx

```diff
@@ -1,57 +1,136 @@
 import Sketch from '@sketch-hq/sketch-file-format-ts';
 import { rgbaToHex } from 'noya-colorpicker';
-import { GridView, sketchColorToRgba } from 'noya-designsystem';
+import { MenuItem } from 'noya-designsystem/src/components/ContextMenu';
+import { ContextMenu, GridView, sketchColorToRgba } from 'noya-designsystem';
 import { SelectionType } from 'noya-state';
-import { memo, useMemo } from 'react';
+import { memo, useMemo, useCallback, Fragment } from 'react';
+import { SwatchGroup, createSwatchTree } from '../../utils/createSwatchTree';
+import { sortBy } from 'noya-utils';
 import ColorSwatch from './ColorSwatch';
+import { delimitedPath } from 'noya-utils';
+
+function flatten(
+  SwatchGroup: SwatchGroup,
```

Comment:
Argument names should usually be lowercase, `swatchGroup`, but I'll change it another time

Reply:
oh I'll keep it in mid !